Sources/SwiftRefactor/ConvertStoredPropertyToComputed.swift

```diff
@@ -0,0 +1,46 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if swift(>=6)
+public import SwiftSyntax
+#else
+import SwiftSyntax
+#endif
+
+public struct ConvertStoredPropertyToComputed: SyntaxRefactoringProvider {
+  public static func refactor(syntax: VariableDeclSyntax, in context: ()) -> VariableDeclSyntax? {
+    guard syntax.bindings.count == 1, let binding = syntax.bindings.first, let initializer = binding.initializer else {
+      return nil
+    }
+
+    let body = CodeBlockItemSyntax(
+      item: .expr(initializer.value)
+    )
+
+    let newBinding =
+      binding
+      .with(\.initializer, nil)
+      .with(
+        \.accessorBlock,
+        AccessorBlockSyntax(
+          leftBrace: .leftBraceToken(trailingTrivia: .space),
+          accessors: .getter(CodeBlockItemListSyntax([body])),
+          rightBrace: .rightBraceToken(leadingTrivia: .space)
+        )
+      )
```

Comment:
Would it be worth checking if the initializer is a closure that is immediately called so that converting stored -> computed -> stored results in the same code even if there are multiple statements in the getter?

Reply:
You mean add a specific condition to the method or create a test case where convert to computer and vice versa?

Reply:
I meant that if you start of with

```swift
var x: Int {
  var intermediate = 1 + 1
  return intermediate + 1
}
```

And then perform the covert to stored property refactoring, you get the following, which is what I would expect

```swift
var x: Int = {
  var intermediate = 1 + 1
  return intermediate + 1
}()
```

But then if you convert that to a stored property again (essentially trying to undo the change without hitting Cmd-Z), you get the following

```swift
var x: Int {
  {
    var intermediate = 1 + 1
    return intermediate + 1
  }()
}
```

And that seems overly complicated. I think in this case we should return to the original code when converting back to a stored property.

Reply:
Resolved this in [eac08a4](https://github.com/swiftlang/swift-syntax/pull/2712/commits/eac08a423e06d1153beacfccb1dad6a6c2dfd62b) by checking for a closure and removing it with saving the trivia.
Not understood the idea about converting back to the original code, how could we do that if we don't even save an original state.

FYI, I remember about squashing commits, I will do it after your review :)